### ISSUE TITLE
IMB8_GPS

### ISSUE THREAD

**Issue by tedmaksym** [jeffreymei/SummerWHOI#2]
This is a good opportunity to start to think about what you expect to be going on, versus what your plot shows.  Remember that this is the position of a buoy drifting on sea ice (it may have been stationary for some time at first, depending on which station it was), so it should only be in the ocean, and should drift relatively smoothly (no big jumps).  A lot of the motion you see in your plot is not realistic.  First, check if you are using the GPS data versus the Iridium locations (which can be way off).  Second, this GPS data can experience some spurious jumps, so you need to watch out for those, but in this case it looks like it could be a plotting issue - the direction of travel looks wrong

**Comment by jeffreymei** [comment]
Oh actually you are right, the direction of travel is Blue to Red (I just assumed, incorrectly, that it would go from red to blue)

**Comment by jeffreymei** [comment]
And yes, this data is from Iridium_Lat and Iridium_Long - but in the _td.txt files they only have this location data - should I try the Icebell_8_GPS.txt file instead? I didn't want to use it in case it was not exactly corresponding to Icebell_8_td.txt.

**Comment by jeffreymei** [comment]
The reason why I think Icebell_8_GPS (which has both iridium and non-iridum lat/long) does not correspond to Icebell_8_td (which only has the iridium lat/long) is because of the first column (the "MOMSN", which is some kind of ID) - the ids that are in the _gps.txt file are not in the _td.txt file, and vice versa - and between the two of them they span all the entries. E.g. _gps has 1,2,3,...,19,20,22,...,39,41,42,42... and skips 21 and 40, while _td has 21, 40, and all the other ones missing from the _gps.txt file. It's as if the _td data values were pulled out of the _gps file somehow? In any case, this means that the overall trajectory of the _gps file should be correct for getting the overall path of the _td file, even if the times will not directly match up. I will plot it and see what it looks like.

**Comment by jeffreymei** [comment]
![screenshot from 2015-07-09 14 17 47](https://cloud.githubusercontent.com/assets/3510683/8603302/53c699ea-2645-11e5-9705-3c74019b29c8.png)
Here you can see that 21 is missing from _gps but present in _td.

**Comment by tedmaksym** [comment]
Yes, you should use the GPS.  The Iridium positions can be wrong be 100s of km. It is based on the time the message is received at each of the satellites and then triangulated, so it depends on how many satellite there are and there positions.  Often it's pretty good, but you can clearly see that many times it is wrong.

You will probably find some bad values in the GPS data, too, but since the buoy won't be moving by more than about 1 km/hour, you can pick these out pretty easily.  For just plotting the track, since it moves pretty slowly compared to the size of your map you could just do a moving median filter on the data with a window of an hour or more, and that would probably eliminate most bas values.

**Comment by tedmaksym** [comment]
It looks to me the GPS is sent every 1200 seconds, or 20 minutes, which sounds right.  The reason there is no MOSM 21 in the GPS is it's in the TD file.  The MOSM number increments each time a message is sent by the Iridium, regardless of what that message contains.

**Comment by jeffreymei** [comment]
![imb8_gps_new](https://cloud.githubusercontent.com/assets/3510683/8603723/12180fbc-2648-11e5-9489-73809c310df6.png)

Yes - new GPS values here are much more accurate!! Here is the new plot.

**Comment by tedmaksym** [comment]
Good.  Obviously the GPS at the start is while the buoy was still on the 
ship.  You'll want to start this from the deployment date, in which case 
you can zoom in the area quite a bit

It would be interesting to see the drift of all of the buoys.

Ted

On 7/9/15 2:38 PM, Jeffrey Mei wrote:
>
> imb8_gps_new 
> <https://cloud.githubusercontent.com/assets/3510683/8603723/12180fbc-2648-11e5-9489-73809c310df6.png>
>
> Yes - new GPS values here are much more accurate!! Here is the new plot.
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/jeffreymei/SummerWHOI/issues/2#issuecomment-120099082>.
>


-- 
Ted Maksym
Assistant Scientist
Woods Hole Oceanographic Institution
Bigelow 110, MS#12
Woods Hole, MA 02536
508-289-3428

**Comment by jeffreymei** [comment]
![imb8_stereo](https://cloud.githubusercontent.com/assets/3510683/8605694/fc8f6bba-2654-11e5-9749-167f81d20158.png)
![imb10_stereo](https://cloud.githubusercontent.com/assets/3510683/8605695/fc8f7e34-2654-11e5-9457-f4d89666feb1.png)
![imb17_stereo](https://cloud.githubusercontent.com/assets/3510683/8605693/fc8ef5ea-2654-11e5-8c3e-fba445c13343.png)
Here they are! There are some more that need a bit of cleaning to get rid of the pre-deployment movement tracks.

**Comment by jeffreymei** [comment]
![imb5_stereo](https://cloud.githubusercontent.com/assets/3510683/8606225/46571dc6-2658-11e5-9bc1-44c1fccf853e.png)
![imb6_stereo](https://cloud.githubusercontent.com/assets/3510683/8606223/4656ae68-2658-11e5-8609-5a0c641deaea.png)
![imb9_stereo](https://cloud.githubusercontent.com/assets/3510683/8606224/4656f04e-2658-11e5-9a7e-ed03825368f0.png)
![imb12_stereo](https://cloud.githubusercontent.com/assets/3510683/8606227/465aca5c-2658-11e5-9095-54070736b96d.png)
![imb16_stereo](https://cloud.githubusercontent.com/assets/3510683/8606226/465a3880-2658-11e5-922a-d6833d2ee008.png)
And here are the others! 

You can close this issue if it is completed and open a new topic to discuss another issue.

**Comment by tedmaksym** [comment]
Ok.  You'll want to clean them up a bit to start them when deployed, and get rid of a couple spurious values. Also, put on a scale so we know what the dates are. 

Where was 12 deployed - was that the Amundsen Sea?  That lived long enough it might be worth plotting that anyway for comparison to these.  It looks like you have a few that lived for long enough that it is worth properly processing the temperature plots. What you will need to do is determine the interfaces and put those on the plots - these are the main thing we want to get out of it at this point so we can calculate melt rates.

**Comment by jeffreymei** [comment]
#12 was deployed between the Amundsen and the Ross Seas, near the Getz ice shelf.

Do you want me to plot, say, a thickness of ice shelf colormapped over the map shown, for the GPS coordinates that we have?

**Comment by tedmaksym** [comment]
No,

when you make more refined plots, it's nice to have the extent of the 
ice shelves on there, but the thickness isn't really relevant for this 
so would be more distracting than anything. What we do want to get from 
these is a record of the ice melt, so we want to refine your estimates 
of the interfaces to get some good numbers of the melt rates.  Then we 
can start doing some real science with this.

I've been working on this for the Arctic, so I'm attaching a pic of what 
it should look like (need to clean this up a bit, but this is the basic 
idea).  Here I've tried 3 methods to get the interface, one from the 
temperature profiles, one from the heating rates, and one from the 
standard deviation of the heating rates. They agree reasonably well; in 
this case the heating rates gave the smoothest results, but I will 
probably refine this a bit to use all three together.  Unfortunately, 
your data may be a bit harder, as in the Arctic case I am seeing surface 
melt water at the base of the ice, which makes the interface stand out a 
bit more.






On 7/10/15 1:22 PM, Jeffrey Mei wrote:
>
> #12 was deployed between the Amundsen and the Ross Seas, near the Getz 
> ice shelf.
>
> Do you want me to plot, say, a thickness of ice shelf colormapped over 
> the map shown, for the GPS coordinates that we have?
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/jeffreymei/SummerWHOI/issues/2#issuecomment-120464646>.
>


-- 
Ted Maksym
Assistant Scientist
Woods Hole Oceanographic Institution
Bigelow 110, MS#12
Woods Hole, MA 02536
508-289-3428

**Comment by jeffreymei** [comment]
To add a pic, just drag and drop into the interface here!

**Comment by tedmaksym** [comment]
![imb17t](https://cloud.githubusercontent.com/assets/13255911/8625237/71727254-270b-11e5-9809-5457bc5f52f0.gif)

**Comment by tedmaksym** [comment]
hmm, doesn't let you import postscript.  Not so fancy after all!

**Comment by jeffreymei** [comment]
colorbar success!  two examples: 
![imb6_stereo](https://cloud.githubusercontent.com/assets/3510683/8627879/20937250-271c-11e5-8cb4-7971805f2f13.png)
![imb8_stereo](https://cloud.githubusercontent.com/assets/3510683/8627880/20968e5e-271c-11e5-8eb1-1cbbcb3c9ef1.png)

**Comment by jeffreymei** [comment]
is this what you meant by trying to find the interfaces? i.e. making the color bar discrete, instead of continuous? In the below plot, I could also merge the two orange/yellow ones, but I thought that might lead to some misleading edges.
![imb8_block](https://cloud.githubusercontent.com/assets/3510683/8657128/d4e27626-2969-11e5-9b8b-49c9a452fcdf.png)
